src/observable/ScalarObservable.ts

```diff
@@ -80,8 +80,8 @@ proto.reduce = function <T, R>(project: (acc: R, x: T) => R, seed?: R): Observab
     return <any>this;
   }
   let result = tryCatch(project)(seed, this.value);
-  if (result === errorObject) {
-    return new ErrorObservable(errorObject.e);
+  if (result as any === errorObject) {
```

Comment:
You mean to type [this](https://github.com/ReactiveX/RxJS/blob/master/src/util/errorObject.ts) declaration of `errorObject` to `any`? If this is the case I don't see any problem. What do you think @kwonoj?